bikeshed/update/manifest.py

```diff
@@ -143,28 +145,54 @@ def updateByManifest(path, dryRun=False):
         lastMsgTime = time.time()
         if newPaths:
             say("Updating {0} file{1}...", len(newPaths), "s" if len(newPaths) > 1 else "")
+
+        updateQueue = queue.Queue()
+        success = True
+        # This worker will download from remote, and update success if it fails.
+        def worker():
+            # Needed to update status and progress.
+            nonlocal success
```

Comment:
Even Python has race conditions between threads, so results should be sent back using a queue also.

Reply:
Good point. There is only 1 result, success is initialized to `True`, and any threads that sees a failure will set it to `False`, and won't touch it otherwise.
Since we use a `.join()` below, any threads that have encountered a failure would have set `success` to `False`, and by the time we read `success` in the main thread, threads would have joined, so the value is consistent. Lmk if my understanding of this is wrong. Thanks!

Reply:
Yeah, since the only effect of `worker()` with the sync world is (a) possibly setting `success` to False, and (b) modifying a unique file, I think we're safe here.

Reply:
And the race on `lastMsgTime` will _probably_ just result in writing the "Updated n/m" message twice. And, of course, the number of files the output claims to have already fetched can actually go down if HTTP responses come back out of order.

Reply:
Ah yup the printing is indeed racy.